Play store and App store Apps profitable


This project trails to analysis the profit of apps produced by a company on the two major mobile phone stores that is Apps store for iOS phones and Play store for Android phones

In [1]:
#importing the necessary module
from csv import reader

In [2]:
#Reading the various file into our program

#Reading the app store dataset of july 2017
#We will read the header as a list and the main dataset as another list

ios_file = open('AppleStore.csv', encoding ='utf8' )
ios_data = reader(ios_file)
ios = list(ios_data)

ios_header = ios[0]
ios_apps = ios[1:]

#reading the android data set file

android_file = open('googleplaystore.csv', encoding = 'utf8')
android_data = reader(android_file)
android = list(android_data)

android_header = android[0]
android_apps = android[1:]

#Explore function which will help us better understand our datasets

def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [3]:
#Printing values form our functions above. We will start with 
#data from the iOS dataser

print(ios_header)
print('\n')
explore_data(ios_apps, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


In [4]:
#Printing data from the android dataset
print(android_header)
print('\n')
explore_data(android_apps, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [5]:
#got the row with an error orr the discussion thread of google play store apps
android_apps[10472]

['Life Made WI-Fi Touchscreen Photo Frame',
 '1.9',
 '19',
 '3.0M',
 '1,000+',
 'Free',
 '0',
 'Everyone',
 '',
 'February 11, 2018',
 '1.0.19',
 '4.0 and up']

In [6]:
#this is a method that returns the index of rows with erros
for row in android_apps:
    header_len = len(android_header)
    row_len = len(row)
    if row_len != header_len:
        print(android_apps.index(row))


10472


In [7]:
del android_apps[10472]

In [8]:
for row in ios_apps:
    header_len = len(ios_header)
    row_len = len(row)
    if row_len != header_len:
        print(ios_apps.index(row))

from the above we noticed the for loop did returned anything implying the apple dataset doesn't contain a role with an error.

We will write a generic funtion below that takes two data sets. that is the main datasets with the header and the header file.

In [9]:
def error_dectetor(dataset, header):
    rows = []
    for row in dataset:
        header_len = len(header)
        row_len = len(row)
        if row_len != header_len:
            rows.append(dataset.index(row))
    return rows


# # Removing duplicate entry from the datasets.

Part One. Proving to the user that the dataset contain duplicate entries. 
##### our default dataset id the android_apps

In [10]:
#A generic funtion to determine duplicate entry in any of the dataset

def duplicate_app(app_name, dataset = android_apps):
    for app in dataset:
        name = app[0]
        if name == app_name:
            print(app)
duplicate_app('Instagram')

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


### determining the number of duplicated apps

In [11]:

def duplicated_apps(dataset = android_apps):
    unique_apps = []
    duplicate_apps = []
    for app in dataset:
        name = app[0]
        if name in unique_apps:
            duplicate_apps.append(name)
        else:
            unique_apps.append(name)
    print('Number of duplicated apps is: ', len(duplicate_apps))

duplicated_apps(ios_apps)

Number of duplicated apps is:  0


In [12]:
duplicate_apps = []
unique_apps = []

for apps in android_apps:
    name = apps[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

## Part two
Building the dictionary to filter out apps which are duplicated. We will use the review column of of our dataset and return the with the highest review.

In [13]:
review_apps = {}
for apps in android_apps:
    review = float(apps[3])
    name = apps[0]
    if name in review_apps and review_apps[name] < review:
        review_apps[name] = review
    elif name not in review_apps:
        review_apps[name] = review
    

In [14]:
len(android_apps)
len(review_apps)

9659

Right now we will be creating two dataset. One contains the dataset without duplicate and the other contain files that have already been added

In [15]:
android_clean = []
already_added = []

for app in android_apps:
    name = app[0]
    n_reviews = float(app[3])
    if (review_apps[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

In [16]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


function to return True is characters belong to the English set False other wise

In [17]:
def is_english(string):
    non_ascii = 0
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
False
True
True


In [18]:
android_english_apps = []
ios_english_apps = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english_apps.append(app)
for app in ios_apps:
    name = app[1]
    if is_english(name):
        ios_english_apps.append(app)
        
explore_data(android_english_apps, 0, 3, True)
print('\n')
explore_data(ios_english_apps, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

## Get free apps in the data sets


To obtain the free apps in the dataset we use the price column and return rows where price == 0.

First we create two empty datasets. 
We then loop thru the data set and append apps which pass the condition. 



In [25]:
android_final = []
ios_final = []

for apps in android_english_apps:
    price = apps[7]
    if price == '0':
        android_final.append(apps)
for apps in ios_english_apps:
    price = apps[4]
    if price == '0.0':
        ios_final.append(apps)
        
print(len(android_final))
print(len(ios_final))

8864
3222


# Most apps by Genre
### part one

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we develop it further.
If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.
Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our data sets.

### Part two

We will begin by building functions to generate the frequency of genre

In [31]:
def freq_table(dataset, index):
    total = 0
    table = {}
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
    
    return table_percentages

def display_freq(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    


### part three.
Displaying the frequency table for the prime_genre

In [32]:
display_freq(ios_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


From our frequency table we see that more than 58% of free english apps on Apps store are games. Entertaiment apps make up about 7.88%. these apps are followed by photo and video apps, education and social networking apps which make up about 10% of total apps.

General impression is apps that do well are apps designed for fun that is Games, photo and video, social networking. Where apps designed for practical purposes don't really do well on apps store. Free practical purpose apps on App store don't really do well as compared to apps menat for fun.

However, we can't possible make a conclusion based on the frequency table. We require further investigation into the datasets before we can recommend an app to the market.

Displaying the frequency distribution for the android data set.
Using the Genre and category column. We will start with the category column.

In [37]:
display_freq(android_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In [34]:
display_freq(android_final, 1) #category column

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

From the above Displayed frequency we noticed apps on play store are more distributions with tolld been the highest with 8.44% follow by entertaiment 6.6%.

Also there is no clear difference between the genre and catagory columns of this dataset. 

Play store doesn't exactly have a dominate app genre or catagory as in the case of Apps store which was predominated by games.

# Most Popular Apps by Genre on the App Store
One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Below, we calculate the average number of user ratings per app genre on the App Store:

In [38]:
ios_genres = freq_table(ios_final, -5)

for genre in ios_genres:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    
    avg_genre_rating = total / len_genre
    print(genre, ': ', avg_genre_rating )
            
        

Finance :  31467.944444444445
Shopping :  26919.690476190477
Travel :  28243.8
Food & Drink :  33333.92307692308
Games :  22788.6696905016
Weather :  52279.892857142855
Lifestyle :  16485.764705882353
Utilities :  18684.456790123455
Entertainment :  14029.830708661417
Book :  39758.5
Sports :  23008.898550724636
Navigation :  86090.33333333333
Health & Fitness :  23298.015384615384
Education :  7003.983050847458
Medical :  612.0
Music :  57326.530303030304
Reference :  74942.11111111111
Catalogs :  4004.0
News :  21248.023255813954
Productivity :  21028.410714285714
Business :  7491.117647058823
Photo & Video :  28441.54375
Social Networking :  71548.34905660378


From the ratings you we will noticed navigations have the highest avg_rating on apps store. 
get more details on any of the genre. will use the News genre. we will print the name and the rating

In [40]:
for app in ios_final:
    if app[-5] == 'News':
        print(app[1], ': ', app[5])

Twitter :  354058
Fox News :  132703
CNN: Breaking US & World News, Live Video :  112886
Reddit Official App: All That's Trending and Viral :  67560
USA TODAY :  61724
ABC News - US & World News + Live Video :  48407
NBC News :  32881
HuffPost - News, Politics & Entertainment :  29107
The Washington Post Classic :  18572
WIRED Magazine :  12074
CBS News - Watch Free Live Breaking News :  11691
The Guardian :  8176
AOL: News, Email, Weather & Video :  5233
SmartNews - Trending News & Stories :  4645
MSNBC :  3692
LotteryHUB :  2417
theSkimm :  1765
Quartz • News in a whole new way :  1267
Lotto Results - Mega Millions Powerball Lottery :  794
TopBuzz: Best Viral Videos, GIFs, TV & News :  692
Ticket Scanner for Powerball & MegaMillions Pool :  581
FOCUS Online - Aktuelle Nachrichten :  373
SPIEGEL ONLINE - Nachrichten :  299
n-tv Nachrichten :  273
CNN Politics :  254
Tagesschau :  233
Fresco — Be a part of the news :  219
News Break - Local & World Breaking News & Radio :  173
OPM Aler

In [41]:
for app in ios_final:
    if app[-5] == 'Education':
        print(app[1], ': ', app[5])

Duolingo - Learn Spanish, French and more :  162701
Guess My Age  Math Magic :  123190
Lumosity - Brain Training :  96534
Elevate - Brain Training and Games :  58092
Fit Brains Trainer :  46363
ClassDojo :  35440
Memrise: learn languages :  20383
Peak - Brain Training :  20322
Canvas by Instructure :  19981
ABCmouse.com - Early Learning Academy :  18749
Quizlet: Study Flashcards, Languages & Vocabulary :  16683
Photomath - Camera Calculator :  16523
iTunes U :  15801
Blackboard Mobile Learn™ :  13567
Star Chart :  13482
Remind: Fast, Efficient School Messaging :  9796
PBS KIDS Video :  8651
Toca Kitchen Monsters :  8062
Toca Hair Salon - Christmas Gift :  8049
Edmodo :  7197
Prodigy Math Game :  6683
Epic! - Unlimited Books for Kids :  6676
ChineseSkill -Learn Mandarin Chinese Language Free :  6077
Google Classroom :  5942
TED :  5782
Khan Academy: you can learn anything :  5459
Got It - Homework Help Math, Chem, Physics Solver :  4903
PowerSchool Mobile :  4547
SkyView® Free - Explor

Analyzing the Eduction genre we can see Duolingo learning app has the heighest number of ratings on Apps store

# Most popuplar app on google play store by Genre

In the previous section we come up with the app profile on App store based using the apps ratings. We will be looking at the google play store data using the number of instals to get our profile. 

But the number of installs isn't precised as google uses values like 10K+ downloads 10M+, ect.

we will need to convent these numbers from string values to float. since we don't need the exact number of instal.

#### Let's look at our display frequency using number of install. 
this will help us see the nature of the install values

In [42]:
display_freq(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


#####  creating a frequency table for our various categories.

In [46]:
android_categories = freq_table(android_final, 1)

for category in android_categories:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            num_installs = app[5]
            num_installs = num_installs.replace('+', '')
            num_installs = num_installs.replace(',', '')
            num_installs = float(num_installs)
            total += num_installs
            len_category += 1
    
    avg_num_installs = total / len_category
    print(category, ': ', avg_num_installs)
            

PHOTOGRAPHY :  17840110.40229885
SOCIAL :  23253652.127118643
SPORTS :  3638640.1428571427
FAMILY :  3695641.8198090694
ENTERTAINMENT :  11640705.88235294
HOUSE_AND_HOME :  1331540.5616438356
EDUCATION :  1833495.145631068
MAPS_AND_NAVIGATION :  4056941.7741935486
NEWS_AND_MAGAZINES :  9549178.467741935
BOOKS_AND_REFERENCE :  8767811.894736841
EVENTS :  253542.22222222222
HEALTH_AND_FITNESS :  4188821.9853479853
PRODUCTIVITY :  16787331.344927534
TRAVEL_AND_LOCAL :  13984077.710144928
FINANCE :  1387692.475609756
FOOD_AND_DRINK :  1924897.7363636363
BUSINESS :  1712290.1474201474
BEAUTY :  513151.88679245283
SHOPPING :  7036877.311557789
LIBRARIES_AND_DEMO :  638503.734939759
COMICS :  817657.2727272727
DATING :  854028.8303030303
TOOLS :  10801391.298666667
WEATHER :  5074486.197183099
LIFESTYLE :  1437816.2687861272
AUTO_AND_VEHICLES :  647317.8170731707
GAME :  15588015.603248259
PERSONALIZATION :  5201482.6122448975
ART_AND_DESIGN :  1986335.0877192982
VIDEO_PLAYERS :  24727872.452

From the average number of installs from the various categories we noticed that communications apps have the highest number of installs on average followed by socail apps. 

we will take a closer look at it to examine which apps have greatly influnces these installs.

In [47]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

In [48]:
for app in android_final:
    if app[1] == 'SOCIAL' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+
